pandas/core/indexing.py

```diff
@@ -1107,6 +1107,8 @@ def _getitem_axis(self, key, axis: int):
             # nested tuple slicing
             if is_nested_tuple(key, labels):
                 locs = labels.get_locs(key)
+                if not len(locs):
+                    raise KeyError(key)
```

Comment:
if this is going to raise (not 100% sure, but leaning towards yes), i think it should happen in MultiIndex.get_locs

Reply:
Could not find any information in the docs, if ``get_locs`` should raise when the combination of both keys is not found in MultiIndex. Would prefer that too, but did not want to break anything, hence my question above. Will try to introduce this there

Reply:
We are raising in ``get_locs`` now. This should be much more stable